tutorials-and-examples/storage/hyperdisk-ml/README.md

```diff
@@ -0,0 +1,222 @@
+
+1. Create a new GCE instance that you will use to hydrate the new Hyperdisk ML disk with data. Note a c3-standard-44 instance is used to provide the max throughput while populating the hyperdisk([Instance to throughput rates](https://cloud.google.com/compute/docs/disks/hyperdisks#performance_limits_for_other_vms)).
+
+```sh
+VM_NAME=hydrator
+MACHINE_TYPE=c3-standard-44
+IMAGE_FAMILY=debian-11
+IMAGE_PROJECT=debian-cloud
+ZONE=us-central1-a
+SNAP_SHOT_NAME=hdmlsnapshot
+PROJECT_ID=myproject
+DISK_NAME=model1
+
+gcloud compute instances create $VM_NAME \
+    --image-family=$IMAGE_FAMILY \
+    --image-project=$IMAGE_PROJECT \
+    --zone=$ZONE \
+    --machine-type=$MACHINE_TYPE
+
+gcloud compute ssh $VM_NAME
+
+```
+Update and authenticate the instance
+
+```sh
+sudo apt-get update
+sudo apt-get install google-cloud-cli
+gcloud init
+gcloud auth login
+
+```
+
+2. Create and attach the disk to the new GCE VM.
+
+```sh
+SIZE=140
+THROUGHPUT=2400
+
+gcloud compute disks create $DISK_NAME --type=hyperdisk-ml \
+--size=$SIZE --provisioned-throughput=$THROUGHPUT  \
+--zone $ZONE
+
+gcloud compute instances attach-disk $VM_NAME --disk=$DISK_NAME --zone=$ZONE 
+```
+
+3. Log into the hydrator VM, format the volume, initiate transfer, and dismount the volume.
+
+```sh
+gcloud compute ssh $VM_NAME
+```
+
+Identify the device name (eg: /dev/nvme0n2) by looking at the output of lsblk. This should correspond to the disk that was attached in the previous step. 
+
+```sh 
+lsblk
+```
+
+Save device name given by lsblk(example nvme0n2)
+```sh
+DEVICE=/dev/nvme0n2
+```
+
+```sh
+GCS_DIR=gs://vertex-model-garden-public-us-central1/llama2/llama2-70b-hf 
+sudo /sbin/mkfs -t ext4 -E lazy_itable_init=0,lazy_journal_init=0,discard $DEVICE
+
+sudo mount $DEVICE /mnt
+gcloud storage cp -r $GCS_DIR /mnt
+sudo umount /mnt
+```
+
+4. Detach disk from the hydrator and switch to READ_ONLY_MANY access mode.
+```sh
+gcloud compute instances detach-disk $VM_NAME --disk=$DISK_NAME --zone=$ZONE
+gcloud compute disks update $DISK_NAME --access-mode=READ_ONLY_MANY  --zone=$ZONE
+```
+
+5. Create a snapshot from the disk to use as a template.
+
+```sh
+gcloud compute snapshots create $SNAP_SHOT_NAME \
+    --source-disk-zone=$ZONE \
+    --source-disk=$DISK_NAME \
+    --project=$PROJECT_ID
+```
+
+6. You now have a hyperdisk ML snapshot populated with your data from Google Cloud Storage. You can delete the hydrator GCE instance and the original disk.
+
+```sh
+gcloud compute instances delete $VM_NAME --zone=$ZONE
+gcloud compute disks delete $DISK_NAME --project $PROJECT_ID --zone $ZONE
+```
+
+7. In your GKE cluster create your Hypedisk ML multi zone and standard Storage Classes. Hyperdisk ML disks are zonal and the Hyperdisk-ml-multi-zone storage class automatically provisions disks in zones where the pods using them are. 
```

Comment:
What does "standard" mean? It may be confusing because there's a "pd-standard" disk type.